src/analyzer/analyzersilence.h

```diff
@@ -13,16 +15,29 @@ class AnalyzerSilence : public Analyzer {
 
     bool initialize(const AnalyzerTrack& track,
             mixxx::audio::SampleRate sampleRate,
-            int totalSamples) override;
-    bool processSamples(const CSAMPLE* pIn, const int iLen) override;
+            SINT totalSamples) override;
+    bool processSamples(const CSAMPLE* pIn, SINT iLen) override;
     void storeResults(TrackPointer pTrack) override;
     void cleanup() override;
 
+    /// returns the index of the first sample in the buffer that is above -60 dB
+    /// or iLen if no sample is found
+    static SINT findFirstSoundInChunk(std::span<const CSAMPLE> samples);
+
+    /// returns the index of the last sample in the buffer that is above -60 dB
+    /// or -1 if no sample is found.
+    static SINT findLastSoundInChunk(std::span<const CSAMPLE> samples);
+
+    /// Returns true if the first sound if found at the given frame and logs a
+    /// warning message if not. This can be uses to detect changes since the
+    /// last analysis run and is an indicator for file edits or decoder
+    /// changes/issues
+    static bool verifyFirstSound(std::span<const CSAMPLE> samples,
+            mixxx::audio::FramePos firstSoundFrame);
+
```

Comment:
IMO these are very easily unit testable, it would be nice to have some of those.

Reply:
Done